### PR TITLE
Add definitions for expand_dimensions to the Makie extension

### LINKED ISSUE #720: Towards a nicer Makie recipe
It occurs to me that dimensional axes are _dimensional_, and we can use Makie's new unit integration to deal with them.

For example, instead of defining conversions in convert_arguments, we might also define `expand_dimensions` for dimensional arrays, so that Makie can pick up on the dimension of the unit.  Defining a custom dim conversion might also allow axis modifications like default labels, without the need to override the basic `plot!` function.

This isn't particularly urgent, but would be (a) nice to have and (b) pave the way for more such integration in the Makie ecosystem.

### PR DESCRIPTION
**Pull request by asinghvi17**
Partially fix #720, fix the issue I was having with plotting Rasters in GeoMakie's meshimage as well.

### CODE REVIEW

File: ext/DimensionalDataMakie.jl [hunk 1]
```diff
@@ -503,7 +510,7 @@ function _keywords2dimpairs(x, y)
     end
 end
 
-_floatornan(A::AbstractArray{<:Union{Missing,<:Real}}) = _floatornan32.(A)
+_floatornan(A::AbstractArray{<:Union{Missing,<:Real}}) = _floatornan64.(A)
```
[the remark this answers is not in the document]
Comment (asinghvi17) [reply ext/DimensionalDataMakie.jl:513]: Yep!

**Comment by asinghvi17** [comment]
Looks like a Plots.jl version conflict causing some form of downgrade in the nightly CI

**Comment by rafaqz** [comment]
Ah right. I've just been ignoring nightly 😅

**Comment by asinghvi17** [comment]
Nice lol.  If all the CI passes then this looks good to merge to me - I can confirm that my GeoMakie issue is resolved with this.

**Review by rafaqz — APPROVED** [review]

**Pull request merged**